src/verilog.y

```diff
@@ -5529,9 +5529,14 @@ property_spec<nodep>:                   // IEEE: property_spec
 pexpr<nodep>:  // IEEE: property_expr  (The name pexpr is important as regexps just add an "p" to expr.)
         //UNSUP: This rule has been super-specialized to what is supported now
         //UNSUP remove below
+                complex_pexpr                           { $$ = $1; }
+        |       expr                                    { $$ = $1; }
+        ;
+complex_pexpr<nodep>:
                 expr yP_ORMINUSGT pexpr                 { $$ = new AstLogOr($2, new AstLogNot($2, $1), $3); }
         |       expr yP_OREQGT pexpr                    { $$ = new AstImplication($2, $1, $3); }
-        |       expr                                    { $$ = $1; }
+        |       yNOT pexpr %prec prNEGATION             { $$ = new AstLogNot($1, $2); }
+        |       '(' complex_pexpr ')'                   { $$ = $2; }
```

Comment:
Do these new rules align with IEEE?  The rules around "expr" are very complicated to avoid shift/reduce etc problems, so need to make sure whatever we add won't cause problems when we handle more complicated expressions in the future.

Reply:
That new rule should be compatible with IEEE, but it doesn't come from it.

It solves the problem of parentheses appearing both in pexprs and exprs which results in ambiguity. For example, in the following pexpr: `(((expr)))` it's unclear to YACC which parens enclose a pexpr and at which level an expr begins.

The rule I wrote enforces YACC to interpret every paren as enclosing an expr, unless it contains something that belongs to a pexpr.

In fact, these both rules will need to be deleted when support for sequences is added, but as far as I know, the full IEEE compliant grammar will also have that ambiguity problem, and then we can use a similar strategy to the one proposed here.

Reply:
Ok, that seems reasonable, please add a comment to the sources saying something similar.